lib/vanagon/project.rb

```diff
@@ -82,6 +82,39 @@ def get_files
       files.flatten.uniq
     end
 
+    # Returns a filtered out set of components only including those
+    # components necessary to build a specific component. This is a
+    # recursive function that will call itself until it gets to a
+    # component with no build requirements
+    #
+    # @param name [String] name of component to add. must be present in configdir/components and named $name.rb currently
+    # @return [Array] array of Vanagon::Component including only those required to build "name"
```

Comment:
This isn't complete; it can also return`nil` (line 94), which is not iterable like an empty array would be. I'd actually return an empty array instead of `nil` to minimize the amount of type checking needed (instead of looking for `#nil?` **and** `#empty?`, you can just look for `#empty?` safely).